src/ucs/datastruct/piecewise_func.h

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_PIECEWISE_FUNC_H_
+#define UCS_PIECEWISE_FUNC_H_
+
+#include <ucs/datastruct/linear_func.h>
+#include <ucs/debug/assert.h>
+#include <stddef.h>
+#include <limits.h>
+
+
+struct ucs_piecewise_segment;
+typedef struct ucs_piecewise_segment ucs_piecewise_segment_t;
+/*
+ * A piecewise func segment which represents linear function on a half-open
+ * interval (end is included, start is not).
+ */
+struct ucs_piecewise_segment {
+    ucs_linear_func_t       func;
+    size_t                  end;
+    ucs_piecewise_segment_t *next;
+};
+
+
+/**
+ * A piecewise function consisting of one or more segments.
+ */
+typedef struct {
+    ucs_piecewise_segment_t *head_segment;
+} ucs_piecewise_func_t;
+
+
+ucs_piecewise_segment_t *_ucs_piecewise_segment_alloc()
```

Comment:
why _?

Reply:
According to datasruct/bitmap.h functions that are supposed to be used only internally starts with underscore